catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,604 @@
+                resultNode);
+        String descriptiveResult = getDataQualityResultValue("mgmp:MGMP_DescriptiveResult",
+                "statement",
+                resultNode);
+        if (StringUtils.isNotEmpty(quantitativeResult)) {
+            return quantitativeResult;
+        } else if (StringUtils.isNotEmpty(descriptiveResult)) {
+            return descriptiveResult;
+        }
+        return null;
+    }
+
+    private String getDataQualityResultValue(String resultType, String resultField,
+            Node resultNode) {
+        if (resultNode.getNodeName()
+                .equals(resultType)) {
+            NodeList resultNodeChildren = resultNode.getChildNodes();
+            for (int p = 0; p < resultNodeChildren.getLength(); p++) {
+                Node childNode = resultNodeChildren.item(p);
+                if (childNode.getNodeName()
+                        .equals(resultField)) {
+                    return childNode.getTextContent()
+                            .trim();
+                }
+            }
+        }
+        return null;
+    }
+
+    private void addMetacardMetadataSecurityOriginatorClassification(
+            final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+        addFieldToMetacardIfPresent(pathValueTracker,
+                metacard,
+                MgmpConstants.METADATA_ORIGINATOR_SECURITY_PATH,
+                Security.METADATA_ORIGINATOR_CLASSIFICATION);
+    }
+
+    private void addMetacardResourceSecurityOriginatorClassification(
+            final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+        addFieldToMetacardIfPresent(pathValueTracker,
+                metacard,
+                MgmpConstants.RESOURCE_ORIGINATOR_SECURITY_PATH,
+                Security.RESOURCE_ORIGINATOR_CLASSIFICATION);
+    }
+
+    private void addMetacardCloudCoverPercentage(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        List<String> cloudCoverage =
+                pathValueTracker.getAllValues(toPath(MgmpConstants.CLOUD_COVERAGE_PATH));
+        if (CollectionUtils.isNotEmpty(cloudCoverage)) {
+            metacard.setAttribute(Isr.CLOUD_COVER, (Serializable) covertStringListToDoubleList(
+                    cloudCoverage));
+        }
+    }
+
+    private void addValuesFromPathToListIfPresent(final XstreamPathValueTracker pathValueTracker,
+            List<String> list, String path, boolean parseCrsInformation) {
+        List<String> values = pathValueTracker.getAllValues(toPath(path));
+        if (CollectionUtils.isNotEmpty(values)) {
+            if (parseCrsInformation) {
+                list.addAll(parseCrsInformationFromWktSpecification(values));
+            } else {
+                list.addAll(values);
+            }
+        }
+    }
+
+    private List<String> parseCrsInformationFromWktSpecification(List<String> crsCodes) {
+        List<String> crsCodeList = new ArrayList<>();
```

Comment:
To avoid the side effects and the complex lambda you have, this can all be replaced with:
```java
return crsCodes.stream()
        .map(crsCode -> StringUtils.substringBetween(crsCode, ",ID[", "]"))
        .filter(StringUtils::isNotEmpty)
        .map(inf -> inf.replace('"', ''))
        .map(inf -> inf.replace(',', ':'))
        .collect(Collectors.toList());
```
Note I also swapped out that `replaceAll("\"", "")` with the simpler `replace('"', '')` and also changed the original `replace(CharSequence, CharSequence)` to its little brother that operates on single characters.